reach

```diff
@@ -1113,6 +1113,196 @@ EOF
+fB9SMMpT3VgoCyXxRk1U2Khg0/4bVAQE90o9Yn99zvVLnlJ3nWYsYf/nBSgNxmdp
+HdrLSX1vs/J1vLXebKGj6M4ik3I7NtNMBANWDYt30b57rKsjL+gsJ26MpIRxKQvb
+JhrV8n1ICzVUl63p+oEse7goZcRiev41r5zkokk4LdAwUm4fogKApQMQnepSCqoz
+9uKD47tCcy8xiYWNLZhGyt/qLwsAvhDjhsJcYzTlTUvEfUBdyQ2cFz1LGMryZaZn
+r5jb3kti1D5GRJrJUVJEwvY5mImGYQ0n8YrjS/EUfEaDH0YwSDmEz4bvyqTw5ZQ9
+cT2U7BiAT3ePtcL64ZzQqENITjfswQXkMsh7XzQcbU4iHlmFjzRe0IwoYH9/qzKn
+tbJeNwgRZMy7r1I7FUnGK0Nd+I6D26tMq/hn7Eusd63ZW+CB2ANQOhKYcuGihVAn
+alEhVMzer6D0LAiuvEvOER9930JZp1yhc2t1mnYOzLcTOIZ5r3y3N/s0i0W3hocb
+g7w2sVOO5KNHdHHKr+rnfbAigAXT1yLbRzubSEk75vawgvnjGdZUv2Z1Jzfu9Dmv
+11RTlJ/v9sG2wtldZngVnGNbWrUyvHIt2vn+aWHuC+fJzXdbS+zIIb0RXEUbfJf+
+eUAYNFcH1d0usHcJh1tFCgOZi+pjmZ+mhH58csA62ehi7qKIkRRzAYVs5Lbi2aKl
+hBqUtQ3rnGVi5iGr7V4u6PUVCHFmdhCROaWWiC19soxY74x1URLTYvLAsD6DYUbi
+XzFSB4p298poxBO5MhZ2bCa7RSEbMGqIGyodym6pQYGXqZkDB9iAxFUFQw5BiT8S
+qrGXSCwmddYkEY1A4x6kVFjeJvtM939J/9xgc61Od/b+aVLbn01KNdtpIuz7dVf7
+bXiukztIjBRRCBVWgyLwnDaSAl0Yf83nbGDYA2DCggvyG23h5ddlpCso2zlbRRem
+wwquYXEXzpvWZGXc6FAa3JA3swJnrAu0fK44KRZ1746vXth8JXkudBbEO0phsXiy
+RsVNufkGeUq6Lv/wUZ5mnOcA4Yt8AIlvvf1LfaF4h4kUnpYE2XR/vWoV0APbL79+
+I7+hfxiM0n0KgwTYLA8ttRVMJaA524Qi/nMRSAIeVOO9s1fgWAf+YQfwbgGPignf
+9xncLdQ5vbXZoRXn0aGgBdhpKFe4IjQzVG3nDE3MPYZ0xBfq0rZ4jrr6fJhvNDDo
+0+91M1XN0SJvit/dnC44LNu45iqB3VqKy78Z/fwJhdh5U9uqEITy64tLsT4aHgz7
+mEMfPZ6zMQVFLWglVFQHSOQBtlJ2qXj50B0p6oxxFPXRjOmTBr0l8yJRmrMBCGLs
+HB0F9qvX8fd+rmsW8MRf4vFaJswJqESPHqE1WCVY8GXzo6NO/Rvhcw3hL50ZdZIK
+344B9D9FD2LNmTQJ9mobKrRvt8YflHVQZWptGRaHS5F+eHTN7zecK+t+IPcjCclV
+0qgN6Rh7QtMWoYHp3XqffigixqqOzi/utoTgwRfG6DCYBqySZexkX2lQuIPrvL2v
+FWLg0p11fAuYWx2boj1uev6OxMZLry1KWWCyjDq3OM6W+qQzso1TRhPta5pya+JP
+TWlMmCyf1RmovfvkGKvpKqgpHzycKMvtUaFR24qFYoz/2v4+blaqYnfsq4MQ8g9P
+PBAOvknaoqav3wi0m9BOeuz4I3qyot/1t3WQCDmXUvyu2DAqxYQdI9gYMClq2Mkl
+9e7dPlXqfFH/IK3R4Sk7zaOnJaFzF58sF1xpZEp45ad9/tZcX1VwpMDV1RXzxzYk
+KroLJ0G8lKRwAvIY1o8K7SnT7GRVCtRtZ8FK7KNJzMAclRtpK+PvIgIVzz1ejs+0
+9RLyfdT+KLvc4hmxeqK99KEykgbZ5k4uWkxcG87ONit51zoHUnhkkGs+U54I/tok
+YgdLh+4Iieyt5DoNKK1n0weM66aytN8Jf44XSLAZT6Slr4xUPMEX9Z0HbULLZc5C
+L++5YrWi0IYutYAANujzeA==
+-----END ENCRYPTED PRIVATE KEY-----
+'
+
+TLS_DEFAULT_PASSPHRASE=rpc-demo
+TLS_SAN=reach-server
+
+do_tls () {
+  TLS_KEY="./tls/$TLS_SAN.key"
+  TLS_CSR="./tls/$TLS_SAN.csr"
+  TLS_CRT="./tls/$TLS_SAN.crt"
+  TLS_EXT="./tls/$TLS_SAN.ext"
+
+  fail_req () {
+    if [ -e "$TLS_KEY" ]; then rm "$TLS_KEY"; fi
+    if [ -e "$TLS_EXT" ]; then rm "$TLS_EXT"; fi
+    exit 1
+  }
+
+  print_dir () {
+    (command -v tree >/dev/null && tree ./tls) \
+      || ls -alh ./tls
+  }
+
+  if [ -e "$TLS_KEY" ] \
+  || [ -e "$TLS_CSR" ] \
+  || [ -e "$TLS_CRT" ] \
+  || [ -e "$TLS_EXT" ]; then
+    printf "Aborting \`reach tls\` due to existing certificate in ./tls directory.\n\n"
+    print_dir
+    exit 1
+  fi
+
+  echo "*** Warning! Certificates generated by \`reach tls\` are only suitable ***"
+  echo "    for development purposes and must not be used in production!"
+  echo
+
+  if command -v openssl >/dev/null; then
```

Comment:
I don't think that you should run this whenever it is available, because if someone is running the tutorial and happens to have `openssl` installed, they will get a different path. I think it should take an argument that is either `generate` or `hardcoded`.

But actually, thinking about it harder... if we tell them not to use the `openssl` generated ones anyways... who cares if they use the ones we bake in? They're both just as bad, so they shouldn't use any of them. This whole path just complexifies things for them.